raster-foundry/raster-foundry
# Add Backsplash Lab Integration

## Overview

This adds an initial analysis service that builds on the usage of Project nodes. It does not add the histogram support that is necessary for full integration as that is going to be dealt with in https://github.com/raster-foundry/raster-foundry/issues/4151

### Checklist


### Demo

#### Old Lab
![old-tileserver](https://user-images.githubusercontent.com/898060/46536866-c36d7d80-c87d-11e8-8594-10f9c7d5d844.png)

#### New Lab
![new-tileserver](https://user-images.githubusercontent.com/898060/46536871-c8323180-c87d-11e8-8202-8a3a068ae3d9.png)

#### Single Band Coloring Still Works Too
![singleband-still-works](https://user-images.githubusercontent.com/898060/46536877-d08a6c80-c87d-11e8-9e61-138fa3b99fc9.png)


## Testing Instructions

There is somewhat of a roundabout way for testing since the lab UI doesn't function without a working histogram endpoint. So the first step is to edit `docker-compose.yml` and set the tile server to the old one:
```
TILE_SERVER_LOCATION=http://localhost:9101
```
 * Create a new analysis with a project that has a histogram (if using a COG)
 * Copy a tile request URL from the network inspector
 * Modify the domain of the request to point to the new tile server (`localhost:8081`) and make request
 * Visualization should match

Closes #3905
